src/main/java/org/dataone/hashstore/filehashstore/FileHashStore.java

```diff
@@ -540,266 +508,66 @@ private ObjectMetadata syncPutObject(
-        long objInfoRetrievedSize = objectInfo.getSize();
-        String objCid = objectInfo.getCid();
-
-        if (objInfoRetrievedSize != objSize) {
-            logFileHashStore.info(
-                "FileHashStore.verifyObject - Object size invalid for cid: " + objCid
-                    + ". Expected size: " + objSize + ". Actual size: " + objInfoRetrievedSize
-            );
-            return false;
-
-        } else if (!digestFromHexDigests.equals(checksum)) {
-            logFileHashStore.info(
-                "FileHashStore.verifyObject - Object content invalid for cid: " + objCid
-                    + ". Expected checksum: " + checksum + ". Actual checksum calculated: "
-                    + digestFromHexDigests + " (algorithm: " + checksumAlgorithm + ")"
-            );
-            return false;
-
-        } else {
-            logFileHashStore.info(
-                "FileHashStore.verifyObject - Object has been validated for cid: " + objCid
-                    + ". Expected checksum: " + checksum + ". Actual checksum calculated: "
-                    + digestFromHexDigests + " (algorithm: " + checksumAlgorithm + ")"
-            );
-            return true;
-        }
-    }
-
     @Override
     public void tagObject(String pid, String cid) throws IOException, PidRefsFileExistsException,
         NoSuchAlgorithmException, FileNotFoundException, InterruptedException {
-        logFileHashStore.debug(
-            "FileHashStore.tagObject - Called to tag cid (" + cid + ") with pid: " + pid
-        );
+        logFileHashStore.debug("Tagging cid (" + cid + ") with pid: " + pid);
         // Validate input parameters
         FileHashStoreUtility.ensureNotNull(pid, "pid", "tagObject");
         FileHashStoreUtility.ensureNotNull(cid, "cid", "tagObject");
         FileHashStoreUtility.checkForEmptyString(pid, "pid", "tagObject");
         FileHashStoreUtility.checkForEmptyString(cid, "cid", "tagObject");
 
-        synchronized (referenceLockedCids) {
-            while (referenceLockedCids.contains(cid)) {
-                try {
-                    referenceLockedCids.wait(TIME_OUT_MILLISEC);
-
-                } catch (InterruptedException ie) {
-                    String errMsg =
-                        "FileHashStore.tagObject - referenceLockedCids lock was interrupted while"
-                            + " waiting to tag pid: " + pid + " and cid: " + cid
-                            + ". InterruptedException: " + ie.getMessage();
-                    logFileHashStore.error(errMsg);
-                    throw new InterruptedException(errMsg);
-                }
-            }
-            logFileHashStore.debug(
-                "FileHashStore.tagObject - Synchronizing referenceLockedCids for cid: " + cid
-            );
-            referenceLockedCids.add(cid);
-        }
+        // tagObject is synchronized with deleteObject based on a `cid`
+        synchronizeReferencedLockedCids(cid);
```

Comment:
May put this `synchronized` line into the `try` block? So far it is good, but in the future, someone may add lines between them. Then it can cause an issue.

Reply:
These have been moved, thank you!